File: backend/app/schemas/dna_config.py
from pydantic import BaseModel, ConfigDict
from typing import Dict, Any, List, Optional
from datetime import datetime
from uuid import UUID

# DNA Question
class DNAQuestionBase(BaseModel):
    question_id: str
    category: str
    question_text: str
    options: List[Dict[str, Any]] = []
    order_index: Optional[str] = None

class DNAQuestionCreate(DNAQuestionBase):
    pass

class DNAQuestionUpdate(BaseModel):
    category: Optional[str] = None
    question_text: Optional[str] = None
    options: Optional[List[Dict[str, Any]]] = None
    order_index: Optional[str] = None

class DNAQuestion(DNAQuestionBase):
    id: UUID
    created_at: Optional[datetime]
    updated_at: Optional[datetime]

    model_config = ConfigDict(from_attributes=True)

# DNA Result Template
class DNAResultTemplateBase(BaseModel):
    template_id: str
    risk_level: str
    recommended_package: Optional[str] = None
    content: Optional[str] = None

class DNAResultTemplateCreate(DNAResultTemplateBase):
    pass

class DNAResultTemplateUpdate(BaseModel):
    risk_level: Optional[str] = None
    recommended_package: Optional[str] = None
    content: Optional[str] = None

class DNAResultTemplate(DNAResultTemplateBase):
    id: UUID
    created_at: Optional[datetime]
    updated_at: Optional[datetime]

    model_config = ConfigDict(from_attributes=True)


# DNA Evaluation
class DNAEvaluationResponse(BaseModel):
    id: UUID
    patient_name: Optional[str] = None
    patient_age: Optional[int] = None
    patient_email: Optional[str] = None
    patient_phone: Optional[str] = None
    overall_score: int
    risk_level: str
    answers: Dict[str, Any]
    report_content: Dict[str, Any]
    status: str
    notes: str
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class DNAEvaluationUpdate(BaseModel):
    status: Optional[str] = None
    notes: Optional[str] = None

File: client/src/api/dna_config.ts
import { API_BASE_URL } from './config';

const DNA_API_URL = `${API_BASE_URL}/dna-config`;

export interface DNAQuestion {
  id?: string;
  question_id: string;
  category: string;
  question_text: string;
  options: any[];
  order_index?: string;
}

export interface DNAResultTemplate {
  id?: string;
  template_id: string;
  risk_level: string;
  recommended_package?: string;
  content?: string;
}

export async function getDNAQuestions(): Promise<DNAQuestion[]> {
  const res = await fetch(`${DNA_API_URL}/questions`);
  if (!res.ok) throw new Error("Failed to fetch DNA questions");
  return res.json();
}

export async function getDNATemplates(): Promise<DNAResultTemplate[]> {
  const res = await fetch(`${DNA_API_URL}/templates`);
  if (!res.ok) throw new Error("Failed to fetch DNA templates");
  return res.json();
}

export async function updateDNAQuestion(questionId: string, data: Partial<DNAQuestion>, token: string): Promise<DNAQuestion> {
  const res = await fetch(`${DNA_API_URL}/questions/${questionId}`, {
    method: "PUT",
    headers: {
      "Content-Type": "application/json",
      "Authorization": `Bearer ${token}`
    },
    body: JSON.stringify(data),
  });
  if (!res.ok) throw new Error("Failed to update DNA question");
  return res.json();
}

export interface DNAEvaluationResponse {
  id: string;
  patient_name: string | null;
  patient_age: number | null;
  patient_email: string | null;
  patient_phone: string | null;
  overall_score: number;
  risk_level: string;
  answers: Record<string, any>;
  report_content: Record<string, any>;
  status: string;
  notes: string;
  created_at: string;
}

export async function getDNAEvaluations(token: string): Promise<DNAEvaluationResponse[]> {
  const res = await fetch(`${DNA_API_URL}/evaluations`, {
    headers: {
      "Authorization": `Bearer ${token}`
    }
  });
  if (!res.ok) throw new Error("Failed to fetch DNA evaluations");
  return res.json();
}

export async function updateDNAEvaluation(evalId: string, data: { status?: string; notes?: string }, token: string): Promise<DNAEvaluationResponse> {
  const res = await fetch(`${DNA_API_URL}/evaluations/${evalId}`, {
    method: "PUT",
    headers: {
      "Content-Type": "application/json",
      "Authorization": `Bearer ${token}`
    },
    body: JSON.stringify(data),
  });
  if (!res.ok) throw new Error("Failed to update DNA evaluation");
  return res.json();
}

